File: src/main/java/no/nav/tag/dittNavArbeidsgiver/clients/altinn/dto/DelegationRequest.java
package no.nav.tag.dittNavArbeidsgiver.clients.altinn.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public class DelegationRequest {
    public String RequestStatus;
    public String OfferedBy;
    public String CoveredBy;
    public String RedirectUrl;
    public String Created;
    public String LastChanged;
    public boolean KeepSessionAlive = true;

    public List<RequestResource> RequestResources;

    @JsonProperty("_links")
    public Links links;

    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class RequestResource {
        public String ServiceCode;
        public Integer ServiceEditionCode;
    }


    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class Links {
        public Link sendRequest;
    }

    @JsonIgnoreProperties(ignoreUnknown = true)
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Link {
        public String href;
    }
}

File: src/main/kotlin/no/nav/arbeidsgiver/min_side/clients/altinn/dto/DelegationRequest.kt
package no.nav.arbeidsgiver.min_side.clients.altinn.dto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
data class DelegationRequest(
    var RequestStatus: String? = null,
    var OfferedBy: String? = null,
    var CoveredBy: String? = null,
    var RedirectUrl: String? = null,
    var Created: String? = null,
    var LastChanged: String? = null,
    val KeepSessionAlive: Boolean = true,
    var RequestResources: List<RequestResource>? = null,
    @field:JsonProperty("_links") var links: Links? = null
) {

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class RequestResource(
        var ServiceCode: String? = null,
        var ServiceEditionCode: Int? = null
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Links(var sendRequest: Link? = null)

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Link(var href: String? = null)
}